src/librbd/ImageCtx.h

```diff
@@ -68,6 +68,7 @@ namespace librbd {
                                         // a format librados can understand
     std::map<librados::snap_t, SnapInfo> snap_info;
     std::map<std::string, librados::snap_t> snap_ids;
+    std::map<cls::rbd::SnapshotNamespace, librados::snap_t> snap_namespace_ids;
```

Comment:
I'll do this refactoring in a separate branch and will show you. If you like it then I'll rebase this PR on top of the refactoring branch.